assembly-ide/src/assembly/bin/che.sh

```diff
@@ -10,20 +10,436 @@
+  (REQUIRED: WIN|MAC) DOCKER_TOOLBOX_INSTALL_PATH   ==> Location of Docker Toolbox
+  (REQUIRED: WIN|MAC) VBOX_MSI_INSTALL_PATH         ==> Location of VirtualBox  
+  (OPTIONAL) CHE_HOME                               ==> Directory where Che is installed
+  (OPTIONAL) CHE_LOCAL_CONF_DIR                     ==> Directory with custom Che .properties files
+  (OPTIONAL) CHE_LOGS_DIR                           ==> Directory for Che output logs"
+
+USAGE="
+
+Usage: 
+  che [-i] [-i:tag] [-p:port] [run | start | stop]
+
+     -i,      --image        Launches Che within a Docker container using latest image
+     -i:tag,  --image:tag    Launches Che within a Docker container using specific image tag
+     -p:port, --port:port    Port that Che server will use for HTTP requests; default=8080
+     -h,      --help         Show this help
+     run                     Starts Che application server in current console
+     start                   Starts Che application server in new console
+     stop                    Stops Che application server"
+
+function usage {
+  echo "$USAGE"
+}
+
+function error_exit {
+  echo "
+$1"
+  echo "$WRONG $CHE_VARIABLES $USAGE"
+  JUMP_TO_END=true
+}
+
+# Run the finish function if exit signal initiated
+trap exit SIGHUP SIGINT SIGTERM
+
+function parse_command_line {
+
+  ### Parse command line parameters
+  USE_DOCKER=false
+  USE_DOCKER_TAG=latest
+  USE_PORT=8080
+  USE_HELP=false
+  USE_SERVER_ACTION=run
+
+  for command_line_option in "$@"
+  do
+  case $command_line_option in
+    -i|--image)
+      USE_DOCKER=true
+      shift # past argument
+    ;;
+    -i:*|--image:*)
+      USE_DOCKER=true
+      USE_DOCKER_TAG="${command_line_option#*:}"
+      shift # past argument=value
+    ;;
+    -p:*|--port:*)
+      USE_PORT="${command_line_option#*:}"
+      shift # past argument=value
+    ;;
+    -h|--help)
+      USE_HELP=true
+      usage
+      return
+      shift # past argument
+    ;;
+    start|stop|run)
+      USE_SERVER_ACTION=${command_line_option}
+      shift # past argument with no value
+    ;;
+    *)
+            # unknown option
+    ;;
+  esac
+  done
+}
+
+function determine_os {
+  # Set OS.  Mac & Windows require VirtualBox and docker-machine.
+  win=false
+  mac=false
+  linux=false
+
+  if [[ "$OSTYPE" == "linux-gnu" ]]; then
+    # Linux
+    linux=true
+  elif [[ "$OSTYPE" == "darwin"* ]]; then
+    # Mac OSX
+    mac=true
+  elif [[ "$OSTYPE" == "cygwin" ]]; then
+    # POSIX compatibility layer and Linux environment emulation for Windows
+    win=true
+  elif [[ "$OSTYPE" == "msys" ]]; then
+    # Lightweight shell and GNU utilities compiled for Windows (part of MinGW)
+    win=true
+  elif [[ "$OSTYPE" == "win32" ]]; then
+    # I'm not sure this can happen.
+    win=true
+  elif [[ "$OSTYPE" == "freebsd"* ]]; then
+    # FreeBSD
+    linux=true
+  else
+    # Unknown.
+    echo We could not detect your operating system.
+    echo Che is unlikely to manage Docker properly.
```

Comment:
I have a general error_exit function to control this.  I go ahead and error out.